lib/refresh_ucf_list.rb

```diff
@@ -0,0 +1,44 @@
+class RefreshUcfList
+  attr_accessor :model_name
+
+  def initialize model_name
+    @model_name = model_name
+  end
+
+  SPECIAL_CHARACTER_LISTS = /[?{}\[\]]/
+
+  def filter_id
+    retrieve_name_columns.each do |name|
+      original_stdout = STDOUT.clone
+      STDOUT.reopen(new_file(name), "w")
+      special_character_records(name).each do |record| 
+        puts "#{record.id}\n"
+      end
+      STDOUT.reopen(original_stdout)
+      puts "Total number of ids for #{name}: #{special_character_records(name).count}"
+    end
+  end
+
+  private
+
+  # Fetch all the column attribute names from the table
+  def fetch_columns
+    @model_name.attribute_names
+  end
+
+  # Retrieve the column attribute where like 'name'
+  def retrieve_name_columns
+    fetch_columns.grep /name/
+  end
+
+  # Retrieve the special character records from the model
+  def special_character_records column_name
+    @model_name.where(column_name.to_sym => SPECIAL_CHARACTER_LISTS)
```

Comment:
This looks good, but when we run it we may run into Mongoid session time-outs.  No need to change this unless we run into the timeout--which is usually expressed by an error message about a lost or closed cursor--but if we need to, we can add the `no_timeout` flag on the pattern seen at https://github.com/FreeUKGen/MyopicVicar/blob/a275d57d278af4a334b5b11aca81d2db45f967c3/lib/tasks/add_day_to_search_queries.rake#L9